src/timer/mod.rs

```diff
@@ -40,25 +39,23 @@ impl<'a> TimerGroup<'a> {
     pub fn start<Context>(
         &mut self,
         timer: &'a Timer<'a>,
-        context: &Context,
+        context: &'a Context,
         callback: fn(context: &Context),
     ) {
         timer.value.remaining_ticks.replace(0);
-        timer
-            .value
-            .context
-            .replace(Some(unsafe { core::intrinsics::transmute(context) }));
         timer
             .value
             .callback
-            .replace(Some(unsafe { core::intrinsics::transmute(callback) }));
+            .replace(Some(Callback::new(context, callback)));
 
         self.timers.push_back(timer);
     }
 
     pub fn run(&mut self) {
         for timer_data in self.timers.iter() {
-            (timer_data.callback.get().unwrap())(timer_data.context.get().unwrap());
+            if let Some(callback) = timer_data.callback.get() {
+                callback.call();
+            }
```

Comment:
Wouldn't `unwrap` here be better? As designed it should be impossible to have a `None` so silently ignoring this seems like a regression.

Reply:
I was cleaning it up and missed that point, though I think using `expect` would be better as we get the same behavior but with an error message that is more descriptive

Reply:
I have not used `expect` before. I like that option.

Reply:
And yeah, this is kind of a weird case. The `Option` is only there so that the client can allocate memory before we know what it needs to store.